static/js/generator.js

```diff
@@ -1,246 +1,251 @@
+            $.post(this.submitResutsUrl+this.data[i].id+"?session="+this.session);
+
+        }
+    },
+
+    /**
+     *	Check selected answer with success condition
+     **/
+    calcSessionTime: function (){
+        var endTime = new Date();
+        var total = endTime.getMinutes()-this.startTime.getMinutes();
+        return (total <= 1)?"1 minute":total + " minutes";
+    },
+
+    /**
+     *	When the ex are done perform an action
+     **/
+    onExSetComplete: function (){
+        var _this = this;
+        var redirect = _this.distractionShieldOriginalDestination();
+        swal({
+                title: "You rock!",
+                text: "That took less than "+ _this.calcSessionTime() + ". practice more?",
+                type: "success",
+                showCancelButton: true,
+                confirmButtonColor: "#7eb530",
+                confirmButtonText: "Let's do it!",
+                cancelButtonText: redirect!=null?"Take me away!":"Go home!",
+                closeOnConfirm: true
+            },
+            function(isConfirm){
+                if(isConfirm){
+                    _this.restart();
+                    return;
+                }
+                _this.terminateGenerator();
+                if (redirect!=null) {
+                    window.location = redirect;
+                }
+            });
+    },
+
+    terminateGenerator: function(){
+        events.off('exerciseCompleted',this.$eventFunc);
+        events.emit('generatorCompleted');
+    },
+    /**
+     *	Loads the HTML general exercise template from static
+     **/
+    createDom: function(){
+        var _this = this;
+        return $.ajax({
+            type: 'GET',
+            dataType: 'html',
+            url: _this.templateURL,
+            data: this.data,
+            success: function(data) {
+                $("#main-content").html(data);
+            },
+            async: true
+        });
+    },
+
+
+    /**
+     *	Ajax get request to the Zeeguu API to get new bookmarks
+     **/
+    getBookmarks: function(){
+        var _this = this;
+        this.loadingAnimation(true);
+        address = this.bookmarksURL+this.size+"?session="+this.session;
+        return $.ajax({
+            type: 'GET',
+            dataType: 'json',
+            url: address,
+            data: this.data,
+            success: function(data) {
+                _this.loadingAnimation(false);
+            },
+            async: true
+        });
+    },
+
+    /**
+     *	Animation used for loading
+     **/
+    loadingAnimation: function(activate){
+        if(activate === true){
+            this.$container.addClass('hide');
+            this.$loader.removeClass('hide');
+        }else{
+            this.$container.removeClass('hide');
+            this.$loader.addClass('hide');
+        }
+    },
+
+    /**
+     * Extraction of redirect url for Distraction Shield
+     **/
+    distractionShieldOriginalDestination: function() {
```

Comment:
Splitting this function into atomic functions with single responsibility would be better.